lib/graphql/execution/interpreter/runtime.rb

```diff
@@ -439,15 +439,10 @@ def evaluate_selection_with_args(kwarg_arguments, field_defn, next_path, ast_nod
         end
 
         def dead_result?(selection_result)
-          r = selection_result
-          while r
-            if r.graphql_dead
-              return true
-            else
-              r = r.graphql_parent
-            end
-          end
-          false
+          # When a result is marked dead, any of its existing children are marked dead
+          # and it accepts no more writes. But this result might have been initialized,
+          # but not yet assigned to a key in its parent. So check the parent too.
```

Comment:
🤔 how can it _have_ a parent but not yet be assigned to a key in its parent? Don't those happen at the same point?

Instead of checking the parent here, should we check "is the parent dead?" when we assign a key to its parent, and proactively mark it dead at that point? Would that solve this? Or have I totally misunderstood?

Reply:
I'm not _exactly_ sure how it works, but without this check, the test added in https://github.com/rmosolgo/graphql-ruby/pull/3505 fails. 

In that test, the first item in a non-null list that encounters an execution error adds an error to the response and the list is marked dead. The later items in the list don't add errors to the response, since the identify that their parent (the non-null list) is marked dead. 

Maybe there's another way to implement it where the `each { ... }` call is halted when the list is marked dead. But I think there'd be some edge cases with promises, etc, where the iterations of `.each { ... }` all return, but then lazy values cause the list to be marked dead. And I doubt we have tests for those...